### DISCUSSION TITLE
Multiple qbit "root" download folders.

### DISCUSSION THREAD

**Discussion by say-no-to-deer-pressure** [StuffAnThings/qbit_manage#125] [Q&A] [answered]
Hi, I love your program!

I'd like to be use the functionality to remove orphaned files, but I have a question regarding the root folder. I have both Sonarr and Radarr set to add a "movies" and "tv" tag, and qbit is set to download the files to the folder designated by their respective tags, so downloads are never placed directly in the root folder - they're placed in in /downloads/tv/ and /downloads/movies, but /downloads/ itself doesn't contain any data. Does qbit_manage search subfolders?

If it doesn't, I suppose a workaround would be to run two instances, and set each root folder to /downloads/tv/ and /downloads/movies?

Thanks!

**Comment by bakerboy448** [comment]
Based on those paths - it sounds like you intentionally don't want hardlinks and want double space for all torrents; why?

**Reply by say-no-to-deer-pressure** [reply]
Everything is hardlinked; maybe my explanation above was oversimplified. This my setup:
On the Unraid side, everything is on one share: /mnt/user/video/
Downloads go here: 
```
/mnt/user/video/downloads/movies
/mnt/user/video/downloads/tv
```

Then after they are downloaded, they are hardlinked to here:
```
/mnt/user/video/movies
/mnt/user/video/tv
```

For qbittorrent and this script, /mnt/user/video is simply set to /video/

**Comment by bakerboy448** [comment, marked as answer]
Based on your layout — qBittorrent saving to `/video/downloads/{movies,tv}`, hardlinked out to `/video/{movies,tv}` — you don't need two root dirs or two instances here. Set:
```yaml
directory:
  root_dir: /video/downloads/
```
The orphaned/noHL scan walks `root_dir` recursively (`get_root_files` in `modules/util.py`), so `downloads/movies` and `downloads/tv` are both covered by one root.

Don't point `root_dir` at `/video/` itself. Orphan detection is `files_under_root_dir - tracked_torrent_files` (`modules/core/remove_orphaned.py`) with no blanket hardlink exemption — only a narrow inode-based guard for very recently modified files. If `root_dir` included `/video/movies` and `/video/tv`, your permanent hardlinked library would be scanned as orphan-deletion candidates.

For genuinely separate root dirs (no common parent), `root_dir` is still single-path — a list is accepted but only the first entry is used (`modules/config.py`). Docs recommend separate qbit_manage instances for that case (Config-Setup.md, `directory:` section). Not your situation, though — `/video/downloads/` as root_dir covers everything you described.

**Reply by say-no-to-deer-pressure** [reply]
> Based on your layout — qBittorrent saving to `/video/downloads/{movies,tv}`, hardlinked out to `/video/{movies,tv}` — you don't need two root dirs or two instances here. Set:
> 
> ```yaml
> directory:
>   root_dir: /video/downloads/
> ```
> 
> The orphaned/noHL scan walks `root_dir` recursively (`get_root_files` in `modules/util.py`), so `downloads/movies` and `downloads/tv` are both covered by one root.
> 
> Don't point `root_dir` at `/video/` itself. Orphan detection is `files_under_root_dir - tracked_torrent_files` (`modules/core/remove_orphaned.py`) with no blanket hardlink exemption — only a narrow inode-based guard for very recently modified files. If `root_dir` included `/video/movies` and `/video/tv`, your permanent hardlinked library would be scanned as orphan-deletion candidates.
> 
> For genuinely separate root dirs (no common parent), `root_dir` is still single-path — a list is accepted but only the first entry is used (`modules/config.py`). Docs recommend separate qbit_manage instances for that case (Config-Setup.md, `directory:` section). Not your situation, though — `/video/downloads/` as root_dir covers everything you described.

Better late than never I suppose 😂
